### PR TITLE
Schema extension for WATERVERSE use case

### PR DESCRIPTION
**Pull request by grigstat**
Extended the schema to incude 2 new properties for a use case in WATERVERSE. I didn't include the new properties in the examples since they are not required for the model.

### CODE REVIEW

**Review by albertoabellagarcia — COMMENTED** [review]
1) The contribution is possibly right, but we need an extension in the examples as well (and matching the changes in the schema) 
2) we need also where this extension is being used (ADOPTER.yaml file, extend with a new register for the use case)
3) Besides this, possibly you want to be credited as author, therefore put the names in the CONTRIBUTORS.yaml and sign the [contribution agreement](https://smartdatamodels.org/index.php/contributor-agreement/)  if you put en email we will create a certificate for you

**Comment by grigstat** [comment]
Thank you for the feedback Alberto, I made the necessary changes.

File: DataQualityAssessment/schema.json
Comment (albertoabellagarcia) [inline DataQualityAssessment/schema.json:1]: {
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "$schemaVersion": "0.0.4",
  "$id": "https://smart-data-models.github.io/dataModel.DataQuality/DataQualityAssessment/schema.json",
  "title": "Smart Data Models - Data Quality Assessment schema",
  "modelTags": "",
  "derivedFrom": "",
  "license": "https://smart-data-models.github.io/dataModel.DataQuality/DataQualityAssessment/LICENSE.md",
  "description": "This entity describes the data quality properties of a measurement, such as temperature.",
  "type": "object",
  "required": [
    "id",
    "type"
  ],
  "allOf": [
    {
      "$ref": "https://smart-data-models.github.io/data-models/common-schema.json#/definitions/GSMA-Commons"
    },
    {
      "$ref": "https://smart-data-models.github.io/data-models/common-schema.json#/definitions/Location-Commons"
    },
    {
      "properties": {
        "type": {
          "type": "string",
          "enum": [
            "DataQualityAssessment"
          ],
          "description": "Property. NGSI Entity type. It has to be DataQualityAssessment"
        },
        "dateCalculated": {
          "type": "string",
          "format": "date-time",
          "description": "Property. Date of the calculated entity defined by the user"
        },
        "outlier": {
          "type": "object",
          "description": "Property. Includes information about the outlier characteristics of the measurement",
          "properties": {
            "isOutlier": {
              "type": "boolean",
              "description": "Property. Determine whether the measurement has been considered an outlier or not"
            },
            "outlierScore": {
              "type": "number",
              "description": "Property. A score indicating the degree of outlierness"
            },
            "methodology": {
              "anyOf": [
                {
                  "type": "string",
                  "minLength": 1,
                  "maxLength": 256,
                  "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$",
                  "description": "Property. Identifier format of any NGSI entity"
                },
                {
                  "type": "string",
                  "format": "uri",
                  "description": "Property. Identifier format of any NGSI entity"
                }
              ],
              "description": "Relationship. Reference to the other entity including AI methodology information"
            }
          }
        },
        "duplicate": {
          "type": "object",
          "description": "Property. Includes information about duplicated data",
          "properties": {
            "isDuplicate": {
              "type": "boolean",
              "description": "Property. Determine whether the measurement is duplicated or not"
            },
            "foundMatches": {
              "type": "array",
              "description": "Property. If this is a duplicate, determine measurements it does match with",
              "items": {
                "type": "string",
                "description": "Property. Every match found in the data"
              }
            }
          }
        },
        "synthetic": {
          "type": "object",
          "description": "Property. Includes information about the origin of the measurement",
          "properties": {
            "isSynthetic": {
              "type": "boolean",
              "description": "Property. Determine whether the measurement has been created synthetically or not"
            },
            "methodology": {
              "anyOf": [
                {
                  "type": "string",
                  "minLength": 1,
                  "maxLength": 256,
                  "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$",
                  "description": "Property. Identifier format of any NGSI entity"
                },
                {
                  "type": "string",
                  "format": "uri",
                  "description": "Property. Identifier format of any NGSI entity"
                }
              ],
              "description": "Relationship. Reference to the other entity including AI methodology information"
            }
          }
        },
        "accuracy": {
          "type": "number",
          "description": "Property. Accuracy measures the maximum systematic numerical error produced in a sensor measurement"
        },
        "timeliness": {
          "type": "number",
          "description": "Property. Average timeliness of the data-stream. Units:'minutes'"
        },
        "completeness": {
          "type": "number",
          "minimum": 0,
          "maximum": 1,
          "description": "Property. Completeness quantifies the number of missed measurements or observations in a given time window. Units:'P1'"
        },
        "precision": {
          "type": "number",
          "description": "Property. Precision measures the standard deviation of a dataset. That is, it measures how close the values in the dataset are to each other"
        },
        "dataQualityAssessmentVariableNames": {
          "type": "array",
          "description": "Property. Array of variable names used to describe the values for each dataQualityAssessmentDomain property",
          "items": {
            "type": "string",
            "description": "Property. Every item in the variable names array"

          }
        },
        "dataQualityAssessmentDomains": {
          "type": "object",
          "description": "Property. Data quality assessment domains with their respective scores",
          "properties": {
            "completeness": {
              "type": "array",
              "description": "Property. Completeness scores for variables",
              "items": {
                "type": "number",
                "description": "Property. Every item in the completeness array",
                "minimum": 0,
                "maximum": 1
              }
            },
            "consistency": {
              "type": "array",
              "description": "Property. Consistency scores for variables",
              "items": {
                "type": "number",
                "description": "Property. Every item in the consistency array",
                "minimum": 0,
                "maximum": 1
              }
            },
            "timeliness": {
              "type": "array",
              "description": "Property. Timeliness scores for variables",
              "items": {
                "type": "number",
                "description": "Property. Every item in the timeliness array",
                "minimum": 0,
                "maximum": 1
              }
            },
            "uniqueness": {
              "type": "array",
              "description": "Property. Uniqueness scores for variables",
              "items": {
                "type": "number",
                "description": "Property. Every item in the uniqueness array",
                "minimum": 0,
                "maximum": 1
              }
            },
            "validity": {
              "type": "array",
              "description": "Property. Validity scores for variables",
              "items": {
                "type": "number",
                "description": "Property. Every item in the validity array",
                "minimum": 0,
                "maximum": 1
              }
            }
          }
        }
      }
    }
  ]
}

this is the correct version, please check the missing definitions that I have included.

**Review by albertoabellagarcia — CHANGES_REQUESTED** [review]
Plesae check the comment in the schema.json with the new version and [make the test again](https://smartdatamodels.org/index.php/data-models-contribution-api/)

**Comment by grigstat** [comment]
I added the missing descriptions, also passing all 4 tests.

**Review by albertoabellagarcia — CHANGES_REQUESTED** [review]
these are the errors when testing the model

 - Missing NGSI type of ['Property', 'Relationship', 'GeoProperty'] in the description of the property on level 2
  timeliness, completeness, consistency, uniqueness, validity
   - Missing NGSI type of ['Property', 'Relationship', 'GeoProperty'] in the description of the property on level 1
  dataQualityAssessmentVariableNames, dataQualityAssessmentDomains

**Comment by grigstat** [comment]
Fixed the issues, should be fine now. Thanks for the guidance!

**Review by albertoabellagarcia — APPROVED** [review]
great contribution

**Pull request merged**